docs/conf.py

```diff
@@ -117,3 +119,85 @@
 html_last_updated_fmt = '%Y/%m/%d'
 html_sourcelink_suffix = ''
 autoclass_content = 'both'
+
+
+# ----------------------------------------------------------------------------------
+# Source code links
+# ----------------------------------------------------------------------------------
+
+def determine_github_branch() -> str:
+    """Determine the GitHub branch name to use for source code links.
+
+    We need to decide whether to use `stable/<version>` vs. `main` for dev builds.
+    Refer to https://docs.github.com/en/actions/learn-github-actions/variables
+    for how we determine this with GitHub Actions.
+    """
+    # If CI env vars not set, default to `main`. This is relevant for local builds.
+    if (
+        "GITHUB_REF_NAME" not in os.environ
+        and "BUILD_SOURCE_BRANCH_NAME" not in os.environ
+    ):
+        return "main"
+
+    # PR workflows set the branch they're merging into.
+    if base_ref := (
+        os.environ.get("GITHUB_BASE_REF")
+        or os.environ.get("SYSTEM_PULL_REQUEST_TARGET_BRANCH_NAME")
+    ):
```

Comment:
Bad leftover code from Azure

```suggestion
    if base_ref := os.environ.get("GITHUB_BASE_REF")
```